### PR TITLE
Update oracle contracts for cosmwasm v0.14

### PR DESCRIPTION
**Pull request by jrriehl**
## Proposed changes

Update oracle and oracle client contracts for cosmwasm 14.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/fetchai/skills/simple_oracle/handlers.py [hunk 1]
```diff
@@ -175,7 +175,7 @@ def _request_init_transaction(self, ledger_id: str, tx_receipt: JSONLike) -> Non
                         "label": "OracleContract",
                         "init_msg": {"fee": str(strategy.initial_fee_deploy)},
                         "gas": strategy.default_gas_deploy,
-                        "amount": 0,
+                        "amount": 1,
```
Comment (5A11) [inline packages/fetchai/skills/simple_oracle/handlers.py:178]: Why this change?

Comment (jrriehl) [reply packages/fetchai/skills/simple_oracle/handlers.py:178]: This was the easiest way to make sure some funds (along with the denomination) were sent to the contract. Otherwise, we would have needed to send the denomination as an extra string parameter. So this is actually what fixed the test!

File: tests/conftest.py [hunk 2]
```diff
@@ -210,7 +210,7 @@
     "b6ef49c3078f300efe2d4480e179362bd39f20cbb2087e970c8f345473661aa5"
 )
 FUNDED_FETCHAI_PRIVATE_KEY_1 = (
-    "f848e125edb124d7752338fdd15825cd031b8c7f38627ec50ccccb7d75ecffdb"
+    "bbaef7511f275dc15f47436d14d6d3c92d4d01befea073d23d0c2750a46f6cb3"
```
Comment (5A11) [inline tests/conftest.py:213]: So `FUNDED_FETCHAI_PRIVATE_KEY_1` and `FUNDED_FETCHAI_ADDRESS_ONE` now correspond with the validator account (corresponding with DEFAULT_FETCH_MNEMONIC) spawned in LocalFetchDockerImage right?

Comment (5A11) [reply tests/conftest.py:213]: Don't we need to update `FUNDED_FETCHAI_PRIVATE_KEY_2` and `FUNDED_FETCHAI_ADDRESS_TWO` below?

File: tests/test_packages/test_skills_integration/test_simple_oracle.py [hunk 3]
```diff
@@ -156,6 +156,10 @@ def test_oracle(
             f"{DEFAULT_FETCH_LEDGER_ADDR}:{DEFAULT_FETCH_LEDGER_REST_PORT}",
         )
 
+        # use alternate port for prometheus connection
+        setting_path = "vendor.fetchai.connections.prometheus.config.port"
+        self.set_config(setting_path, 9091, type_="int")
```
Comment (5A11) [inline tests/test_packages/test_skills_integration/test_simple_oracle.py:161]: What was the port used before for prometheus connection?

File: tests/conftest.py (hunk 2 above)
Comment (jrriehl) [reply tests/conftest.py:213]: Yes, this key now corresponds to the validator on the local node running in docker. We send funds from this account to the other address, so no need to change the other key / address.

File: tests/test_packages/test_skills_integration/test_simple_oracle.py (hunk 3 above)
Comment (jrriehl) [reply tests/test_packages/test_skills_integration/test_simple_oracle.py:161]: It was `9090` which is exposed in the new 'fetchd` container, so this was just to remove that conflict.

**Review by MissingNO57 — APPROVED** [review]
LGTM

**Review by 5A11 — APPROVED** [review]
LGTM
